src/main/Scheduler.java

```diff
@@ -0,0 +1,151 @@
+package main;
+import java.time.LocalTime;
+import java.util.ArrayList;
+
+import events.ArrivalEvent;
+import events.FloorEvent;
+import states.ActiveState;
+import states.IdleState;
+import states.SchedulerState;
+
+public class Scheduler implements Runnable {
+	private ArrayList<FloorEvent> floorEvents;
+	private ArrayList<ArrivalEvent> arrivalEvents;
+	private ArrayList<DestinationEvent> destinationEvents; 
+	private MiddleMan middleManFloor;
+	private MiddleMan middleManElevator;
+	private SchedulerState currentState; 
+	
+	
+	/**
+	 * Public constructor to create Scheduler object and instantiate instance variables
+	 * @param middleMan Object to hold and pass events to/from the floor
+	 * @param middleMan2 Object to hold and pass events to/from the elevator
+	 */
+	public Scheduler(MiddleMan middleManFloor, MiddleMan middleManElevator) {
+		this.floorEvents = new ArrayList<FloorEvent>();
+		this.arrivalEvents = new ArrayList<ArrivalEvent>();
+		this.destinationEvents = new ArrayList<DestinationEvent>(); 
+		this.middleManFloor = middleManFloor;
+		this.middleManElevator = middleManElevator;
+		this.currentState = new IdleState(this); 
+	}
+	
+	/**
+	 * Used to communicate with Floor Subsystem and Elevator to receive and send events.
+	 */
+	@Override
+	public void run() {
+		while(true) {
+			
+			ArrivalEvent arrivalEvent; 
+			DestinationEvent destinationEvent; 
+			SchedulerEvent schedulerEvent; 
+			FloorEvent floorEvent = getFloorEvent(), currentFloorEvent;
+			boolean floorEventFlag = false, destinationEventFlag = false, keepGoing = false; 
+			
+			currentState = currentState.handleFloorEvent(floorEvent);
+			currentState = currentState.handleArrivalEvent();
+			currentState = currentState.handleDestinationEvent(destinationEvent);
+			
+			if(currentState.getClass() == ActiveState.class) {
+				if (!floorEvents.isEmpty()) {
+					//TO:DO Check if needed. 
+					floorEvent = floorEvents.get(floorEvents.size() - 1);

```

Comment:
wait why are you LastIn First out? Shouldn't it be first in first out?

Reply:
fixed

Reply:
im confused, why are checking `&& !elevatorKeepsGoing` and for the above comment, `floorEvent = floorEvents.get(floorEvents.size() - 1);` is grabbing the event that was last added into the list to put in the middleman, shouldn't it be floorEvent = floorEvents.get(0); ?

Reply:
done